internal/attestation/renderer/renderer.go

```diff
@@ -172,3 +184,105 @@ func getPassFromTerm(confirm bool) ([]byte, error) {
 	}
 	return pw1, nil
 }
+
+func (ab *AttestationRenderer) getSigner(ctx context.Context) (sigstoresigner.Signer, error) {
+	var (
+		signer sigstoresigner.Signer
+		err    error
+	)
+
+	if ab.signingKeyPath != "" {
+		ab.logger.Debug().Str("path", ab.signingKeyPath).Msg("loading key")
+		signer, err = signature.SignerFromKeyRef(context.Background(), ab.signingKeyPath, getPass)
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// key is not provided, let's create one
+		ab.logger.Info().Msg("key not provided, running in key-less mode")
+		signer, err = ab.keyLessSigner(ctx)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return sigdsee.WrapSigner(signer, "application/vnd.in-toto+json"), nil
+}
+
+type certificateRequest struct {
+	PrivateKey *ecdsa.PrivateKey
+	// CertificateRequestPEM contains the signed public key and the CSR metadata
+	CertificateRequestPEM []byte
+}
+
+type Signer struct {
+	sigstoresigner.SignerVerifier
+	Cert  string
+	Chain []string
+}
+
+func (ab *AttestationRenderer) keyLessSigner(ctx context.Context) (sigstoresigner.Signer, error) {
+	request, err := ab.createCertificateRequest()
+	if err != nil {
+		return nil, err
+	}
+	certs, err := ab.certFromChainloop(ctx, request)
+	if err != nil {
+		return nil, err
+	}
+	sv, err := sigstoresigner.LoadECDSASignerVerifier(request.PrivateKey, crypto.SHA256)
+	if err != nil {
+		return nil, err
+	}
+
+	var chain []string
+	if len(certs) > 1 {
+		chain = append(chain, certs[1:]...)
+	}
+
+	return &Signer{
+		Cert:           certs[0],
+		Chain:          chain,
+		SignerVerifier: sv,
+	}, nil
+}
+
+func (ab *AttestationRenderer) createCertificateRequest() (*certificateRequest, error) {
```

Comment:
as I said above, this feels more like a method of a new ChainloopSigner concrete type than of renderer.